### ISSUE TITLE
There is no way to delete a component from a design.

### ISSUE THREAD

**Issue by ekolis** [ekolis/stars-nova#31]
**Describe what is missing**

I can't figure out how to delete a component from a design. It can't be dragged back to the component library and right clicking and double clicking do nothing. I assume this was possible in Stars!?

**Screenshots**
N/A

**Comment by marhawkman** [comment]
yeah it was.  You can REPLACE it though.  For example: place a Long Hump 6 engine on a ship then place a Daddy Long Legs 7 on the ship. it DOES at least let you swap it out.

trying to place specific numbers of components(for min-max reasons) doesn't work well.  It has the basic functionality down, but initial placement of the item is always 1 more than what you want.  Oh wait... if you have more than one of an item you CAN drag out the extras, but not the first one.  hunh, interesting.  Makes me think Nova has quantity coded weirdly.

Guesswork:
placing an object sets what is in the slot.  This is treated as quantity one.  any actual quantity you add is thus 1+X  
So if I shift+drag(which does work) to move 10 at a time setting the slot to "Laser" is 1 + 10 since I drug 10 Lasers.
